File: eCarService.Model/Offer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCarService.Model
{
    public class Offer
    {
        public int OfferId { get; set; }
        public string Name { get; set; }
        public decimal? Price { get; set; }
        public string Status { get; set; }
        public byte[] Image { get; set; }
        public int? CarServiceId { get; set; }
        public virtual ICollection<CarBrandOffer> CarBrandOffers { get; set; }
        public virtual ICollection<OfferPart> OfferParts { get; set; }

        public List<Part> Parts { get; set; }
        public List<CarBrand> CarBrands { get; set; }

        public string PartNames => String.Join(", ", Parts?.Take(3).Select(x => x?.Name));
        public string CarBrandNames => String.Join(", ", CarBrands?.Take(3).Select(x => x?.Name));

    }
}

File: ecarservice_mobile/lib/model/offer.dart
import 'package:flutterv1/model/brand.dart';
import 'package:flutterv1/model/part.dart';
import 'package:flutterv1/screens/offers/offers_screen.dart';

import 'package:json_annotation/json_annotation.dart';
part 'offer.g.dart';

@JsonSerializable()
class Offer {
  int? offerId;
  String? name;
  String? status;
  String? image;
  double? price;
  List<Part>? parts;
  List<CarBrand>? carBrands;
  String? partNames;
  String? carBrandNames;

  List<String>? threeParts;
  List<String>? threeBrands;

  Offer() {}

  factory Offer.fromJson(Map<String, dynamic> json) => _$OfferFromJson(json);

  /// Connect the generated [_$OfferToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$OfferToJson(this);
}
